packages/ethereum/contracts/test/Crypto.t.sol

```diff
@@ -0,0 +1,273 @@
+    ];
+
+    CurvePoint[10] memory points = [
+      CurvePoint(0x74519ef88b32b425a095e4ebcc84d81b64e9e2c2675340a720bb1a1857b99f1e,0xc174fa322ab7c192e11748beed45b508e9fdb1ce046dee9c2cd3a2a86b410936),
+      CurvePoint(0x44548adb1b399263ded3510554d28b4bead34b8cf9a37b4bd0bd2ba4db87ae63,0x96eb8e2faf05e368efe5957c6167001760233e6dd2487516b46ae725c4cce0c6),
+      CurvePoint(0x07dd9432d426845fb19857d1b3a91722436604ccbbbadad8523b8fc38a5322d7,0x604588ef5138cffe3277bbd590b8550bcbe0e523bbaf1bed4014a467122eb33f),
+      CurvePoint(0xe9ef9794d15d4e77dde751e06c182782046b8dac05f8491eb88764fc65321f78,0xcb07ce53670d5314bf236ee2c871455c562dd76314aa41f012919fe8e7f717b3),
+      CurvePoint(0x576d43ab0260275adf11af990d130a5752704f79478628761720808862544b5d,0x643c4a7fb68ae6cff55edd66b809087434bbaff0c07f3f9ec4d49bb3c16623c3),
+      CurvePoint(0xf89d6d261a5e00fe5cf45e827b507643e67c2a947a20fd9ad71039f8b0e29ff8,0xb33855e0cc34a9176ead91c6c3acb1aacb1ce936d563bc1cee1dcffc806caf57),
+      CurvePoint(0x9c91513ccfe9520c9c645588dff5f9b4e92eaf6ad4ab6f1cd720d192eb58247a,0xc7371dcd0134412f221e386f8d68f49e7fa36f9037676e163d4a063fbf8a1fb8),
+      CurvePoint(0x10fee3284d7be6bd5912503b972fc52bf4761f47141a0015f1c6ae36848d869b,0x0b163d9b4bf21887364332be3eff3c870fa053cf508732900fc69a6eb0e1b672),
+      CurvePoint(0xb32b0ab55977b936f1e93fdc68cec775e13245e161dbfe556bbb1f72799b4181,0x2f5317098360b722f132d7156a94822641b615c91f8663be69169870a12af9e8),
+      CurvePoint(0x148f98780f19388b9fa93e7dc567b5a673e5fca7079cd9cdafd71982ec4c5e12,0x3989645d83a433bc0c001f3dac29af861f33a6fd1e04f4b36873f5bff497298a)  
+    ];
+
+    for (uint i = 0; i < u.length; i++) {
+      (uint256 mapped_x, uint256 mapped_y) = crypto.map_to_curve_simple_swu(u[i]);
+      (uint256 p_x, uint256 p_y) = crypto.mapPoint(mapped_x, mapped_y);
+      assertEq(p_x, points[i].x);
+      assertEq(p_y, points[i].y);
+    }
+  }
+
+  function testHashToCurve() public {
+    bytes memory DST = "QUUX-V01-CS02-with-secp256k1_XMD:Keccak256_SSWU_RO_";
+
+    // test strings taken from https://www.ietf.org/archive/id/draft-irtf-cfrg-hash-to-curve-16.html#appendix-J.8.1
+    bytes[5] memory testStrings = [
+      bytes(""),
+      bytes("abc"),
+      bytes("abcdef0123456789"),
+      bytes("q128_qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqq"),
+      bytes("a512_aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa")
+    ];
+
+    // Generated with Rust implementation
```

Comment:
Would be nice to include the chunk of the Rust code in the comments here that was used to generate these.

Reply:
see line 186